team-validator.js

```diff
@@ -18,17 +18,18 @@ class Validator {
 		this.tools = Tools.mod(this.format);
 	}
 
-	validateTeam(team) {
+	validateTeam(team, removeNicknames) {
 		let format = Tools.getFormat(this.format);
-		if (format.validateTeam) return format.validateTeam.call(this, team);
-		return this.baseValidateTeam(team);
+		if (format.validateTeam) return format.validateTeam.call(this, team, removeNicknames);
+		return this.baseValidateTeam(team, removeNicknames);
 	}
 
-	prepTeam(team) {
-		return PM.send(this.format.id, team);
+	prepTeam(team, removeNicknames) {
+		removeNicknames = removeNicknames ? '1' : '0';
+		return PM.send(this.format.id, removeNicknames, team);
 	}
 
-	baseValidateTeam(team) {
+	baseValidateTeam(team, removeNicknames) {
```

Comment:
I'd say to implement `removeNicknames` here rather than in `validateSet`

mostly so it applies even if `format.validateSet` overrides